packages/kbn-utility-types/index.ts

```diff
@@ -98,3 +98,12 @@ export type PublicKeys<T> = keyof T;
  * Returns an object with public keys only.
  */
 export type PublicContract<T> = Pick<T, PublicKeys<T>>;
+
+export type DeepPartialObject<T> = T extends any[]
```

Comment:
1. why does not it handle an array of objects? 
2. `utility-types` is already imported in this file, let's reuse `DeepPartial` from it.

Reply:
The `DeepPartial` from `utility-types` makes the array elements themselves optional, which is not I wanted. For example `DeepPartial<Array<string>>` resolves to `Array<string | undefined>`. I was planning on adding a test and JSDoc here to explain the purpose of this. If it's too narrow of a use-case, I can move this into my implementation.

Reply:
Removed, not needed anymore